File: pipecat_flutter_ios/ios/pipecat_flutter_ios/Sources/pipecat_flutter_ios/PipecatStreamHandlers.swift
//
//  PipecatStreamHandlers.swift
//  Pods
//
//  Created by Kirk Charles Niverba on 2/1/26.
//

class TimelineEventStreamHandler: TimelineEventsStreamHandler {
  private var eventSink: PigeonEventSink<TimelineEvent>?
  
  override func onListen(withArguments arguments: Any?, sink: PigeonEventSink<TimelineEvent>) {
    self.eventSink = sink
  }
  
  override func onCancel(withArguments arguments: Any?) {
    self.eventSink = nil
  }
  
  func sendEvent(_ event: TimelineEvent) {
    eventSink?.success(event)
  }
}

class LocalAudioLevelHandler: LocalAudioLevelStreamHandler {
  private var sink: PigeonEventSink<AudioLevel>?
  
  override func onListen(withArguments arguments: Any?, sink: PigeonEventSink<AudioLevel>) {
    self.sink = sink
  }
  
  override func onCancel(withArguments arguments: Any?) {
    self.sink = nil
  }
  
  func sendLevel(_ level: Double) {
    sink?.success(AudioLevel(level: level))
  }
}

class RemoteAudioLevelHandler: RemoteAudioLevelStreamHandler {
  private var sink: PigeonEventSink<AudioLevel>?
  
  override func onListen(withArguments arguments: Any?, sink: PigeonEventSink<AudioLevel>) {
    self.sink = sink
  }
  
  override func onCancel(withArguments arguments: Any?) {
    self.sink = nil
  }
  
  func sendLevel(_ level: Double) {
    sink?.success(AudioLevel(level: level))
  }
}

File: pipecat_flutter_android/android/src/main/kotlin/com/kcniverba/PipecatStreamHandlers.kt
package com.kcniverba

import com.kcniverba.pipecat_flutter_android.*

class TimelineEventHandlerImpl : TimelineEventsStreamHandler() {
    private var eventSink: PigeonEventSink<TimelineEvent>? = null

    override fun onListen(p0: Any?, sink: PigeonEventSink<TimelineEvent>) {
        this.eventSink = sink
    }

    override fun onCancel(p0: Any?) {
        this.eventSink = null
    }

    fun sendEvent(event: TimelineEvent) {
        eventSink?.success(event)
    }
}

class LocalAudioLevelHandlerImpl : LocalAudioLevelStreamHandler() {
    private var sink: PigeonEventSink<AudioLevel>? = null

    override fun onListen(p0: Any?, sink: PigeonEventSink<AudioLevel>) {
        this.sink = sink
    }

    override fun onCancel(p0: Any?) {
        this.sink = null
    }

    fun sendLevel(level: Double) {
        sink?.success(AudioLevel(level = level))
    }
}

class RemoteAudioLevelHandlerImpl : RemoteAudioLevelStreamHandler() {
    private var sink: PigeonEventSink<AudioLevel>? = null

    override fun onListen(p0: Any?, sink: PigeonEventSink<AudioLevel>) {
        this.sink = sink
    }

    override fun onCancel(p0: Any?) {
        this.sink = null
    }

    fun sendLevel(level: Double) {
        sink?.success(AudioLevel(level = level))
    }
}
